src/app/confirmation/[confirmationKey]/page.js

```diff
@@ -0,0 +1,42 @@
+"use client";
+
+import { useEffect, useState } from 'react';
+import { useRouter, useParams } from "next/navigation";
+import { useMutation } from "react-query";
+import { Error } from "@/app/activities/[activityId]/edit/components/formikFields";
+import { Container, CircularProgress } from '@mui/material';
+import { emailConfirmation } from "@/app/api.mjs";
+
+export default function EmailConfirmation({params}) {
+  const router = useRouter();
+  const [loading, setLoading] = useState(true);
```

Comment:
loading is provided by react-query useMutation.

Reply:
mutation.isLoading - is not the same visual effect. CircularProgress shows shorter than when I do it with state

Reply:
in this case CircularProgress not visible to the user